test/unit/BondingManager.js

```diff
@@ -322,6 +313,80 @@ contract("BondingManager", accounts => {
+                await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound+1)
+                await fixture.roundsManager.execute(bondingManager.address, functionSig("setCurrentRoundTotalActiveStake()"))
+            })
+
+            it("adds a new transcoder to the pool", async () => {
+                await bondingManager.bond(1000, transcoder2, {from: delegator})
+                const firstTranscoder = await bondingManager.getFirstTranscoderInPool()
+                const firstStake = await bondingManager.transcoderTotalStake(firstTranscoder)
+                const secondTranscoder = await bondingManager.getNextTranscoderInPool(firstTranscoder)
+                const secondStake = await bondingManager.transcoderTotalStake(secondTranscoder)
+                const firstDel = await bondingManager.getDelegator(firstTranscoder)
+                const secondDel = await bondingManager.getDelegator(secondTranscoder)
+                assert.equal(firstTranscoder, transcoder1)
+                assert.equal(firstStake.toString(), firstDel.delegatedAmount.toString())
+                assert.equal(secondTranscoder, transcoder2)
+                assert.equal(secondStake.toString(), secondDel.delegatedAmount.toString())
+            })
+
+            describe("evicts a transcoder from the pool", () => {
+                it("last transcoder gets evicted and new transcoder gets inserted", async () => {
+                    bondingManager.TranscoderEvicted({}).on("data", e => {
+                        assert.equal(e.returnValues.transcoder, transcoder0)
+                    })
+                    await bondingManager.bond(2000, transcoder2, {from: delegator})
+                    await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound + 2)
+                    assert.isTrue(await bondingManager.isActiveTranscoder(transcoder2))
+                    assert.isTrue(await bondingManager.isActiveTranscoder(transcoder1))
+                    assert.isFalse(await bondingManager.isActiveTranscoder(transcoder0))
+                    assert.equal((await bondingManager.transcoderTotalStake(transcoder2)).toString(), "2500")
+                })
+
+                it("sets deactivationRound for the inserted transcoder to the max possible round number", async () => {
+                    await bondingManager.bond(2000, transcoder2, {from: delegator})
+                    await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound + 2)
+                    assert.equal((await bondingManager.getTranscoder(transcoder2)).deactivationRound, 2**256-1)
+                })
+
+                it("sets the deactivationRound for the evicted transcoder to the next round", async () => {
+                    await bondingManager.bond(2000, transcoder2, {from: delegator})
+                    await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound + 2)
```

Comment:
Don't think we need this mock if we're just checking if the `deactivationRound` was set properly.

Reply:
fixed in 7fddb5f